app/controllers/admin/schools/cohort_2020_controller.rb

```diff
@@ -19,9 +19,16 @@ def new
       end
 
       def create
-        @user = User.new(params.require(:user).permit(:full_name, :email))
+        @user = User.find_or_initialize_by(params.require(:user).permit(:email))
+        @user.full_name = params.dig(:user, :full_name)
+
         render :new and return unless @user.valid?
 
+        if @user.early_career_teacher?
+          @user.errors.add(:base, "A user with this email address is currently participating as an ECT at school with urn #{@user.teacher_profile.current_ecf_profile.school.urn}")
```

Comment:
Should error messages be in the i18n file now?

Reply:
Yes, probably. What does that give us again?

Reply:
All the error messages in one place primarily, should a content person ever need to review again, @ericaporter recently did some refactoring to move them all together. I know this is an admin page though.